perllib/FixMyStreet/Cobrand/Default.pm

```diff
@@ -814,12 +814,21 @@ cobrand's area_types_children type.
 =cut
 
 sub fetch_area_children {
-    my ($self, $area_id, $all_generations) = @_;
+    my ($self, $area_ids, $all_generations) = @_;
 
-    return FixMyStreet::MapIt::call('area/children', $area_id,
-        type => $self->area_types_children,
-        $all_generations ? (min_generation => 1) : (),
-    );
+    $area_ids = [ $area_ids ] unless ref $area_ids eq 'ARRAY';
+
+    my %all_children;
+
+    foreach my $area_id (@$area_ids) {
```

Comment:
You can call children with an array ref of IDs (e.g. `call('area/children', $area_ids)`) and that should get you all the results you need with one call.

Reply:
I tried doing that but it wasn't returning any results. Spoke to Dave about it and it seems Mapit doesn't have code for looking up children for multiple areas. Unless I've missed something?

https://mapit.mysociety.org/areas/164185,164186/children?type=DIW,LBW,MTW,UTE,UTW,LGE,CED,COP&generation=42

(Not sure what's actually handling that route)

I did briefly start looking at changing Mapit, but seeing as there aren't many bodies with two (or more) areas this seemed like the simpler solution.

Reply:
Oh, I'm just talking nonsense then, sorry!